lib/proxy.js

```diff
@@ -154,16 +154,32 @@ class JWProxy {
     throw new Error(`Didn't know what to do with response code ${response.statusCode}`);
   }
 
+  getSessionIdFromUrl (url) {
+    const sessionIdRe = new RegExp('/session/([^/]+)');
+    if (sessionIdRe.test(url)) {
+      const match = sessionIdRe.exec(url);
+      return match[1];
+    }
+    return false;
+  }
+
   async proxyReqRes (req, res) {
     let [response, body] = await this.proxy(req.originalUrl, req.method, req.body);
     res.headers = response.headers;
     res.set('Content-type', response.headers['content-type']);
     // if the proxied response contains a sessionId that the downstream
     // driver has generated, we don't want to return that to the client.
-    // Instead, return the id for the current session
+    // Instead, return the id from the request or from current session
     body = safeJson(body);
-    if (body && body.sessionId && this.sessionId) {
-      body.sessionId = this.sessionId;
+    if (body && body.sessionId) {
+      const reqSessionId = this.getSessionIdFromUrl(req.originalUrl);
+      if (reqSessionId) {
+        log.info(`Replacing sessionId ${body.sessionId} with ${reqSessionId}`);
+        body.sessionId = reqSessionId;
+      } else if (this.sessionId) {
+        log.info(`Replacing sessionId ${body.sessionId} with ${this.sessionId}`);
+        body.sessionId = this.sessionId;
```

Comment:
The logic behind this is borderline: 
- If the chromedriver session never changes you don't need this. 
- If the chromedriver session changes why we don't propagate it upstream?